derive schema for oranda.json

Gankra opened the pull request:
* Had to replace LinkedHashMap with IndexMap to get derive support (fine)
* Derived schemars::JsonSchema for everything
* TEMP HACK TO DEMO: emit the schema at the end of `oranda build`

Gankra committed:
feat(schema): add json schema for oranda.json

This adds a hidden `oranda config-schema` command which prints the Json Schema
for oranda.json to stdout. It uses schemars' derive(JsonSchema). To allow for
this to work, our uses of LinkedHashMap have been replaced with IndexMap, which
is supported by their derives.

Gankra commented:
main outstanding issue is when we produce the schema and how we surface/archive/publish it

ashleygwilliams reviewed (commented):
this looks good to me - can we add the ci for publishing a release for the schema?

Gankra committed:
feat(schema): add oranda-config-schema.yml

This waits for the Release CI workflow to complete and then runs oranda config-schema and uploads
the result as oranda-config-schema.json
